dnf-behave-tests/fixtures/gpgkeys/sign.sh

```diff
@@ -24,34 +24,43 @@ for KEY_NAME in $KEYSPECS; do
     KEY_DIR="${DIR}/keys/${KEY_NAME}"
     mkdir "${KEY_DIR}"
 
+    # workaround for gpgme unable to handle long paths
+    # * create a temp directory /tmp/<tempdir>
+    # * symlink key directory -> /tmp/<tempdir>/gpghome
+    TMP_DIR=$(mktemp -d)
+    TMP_KEY_DIR="${TMP_DIR}/gpghome"
+    ln -s "${KEY_DIR}" "${TMP_KEY_DIR}"
+
     # create key (without password, without expire)
-    HOME=${KEY_DIR} gpg2 --batch --passphrase '' --quick-gen-key "${KEY_NAME}" default default 0
+    HOME=${TMP_KEY_DIR} gpg2 --batch --passphrase '' --quick-gen-key "${KEY_NAME}" default default 0
 
     if [ "${USE_SIGN_SUBKEY}" = "1" ]; then
         # add sign subkey
-        KEY_ID=$(HOME=${KEY_DIR} gpg2 --list-keys --with-colons "${KEY_NAME}"  | grep '^fpr:' | head -n 1 | cut -d : -f 10)
-        HOME=${KEY_DIR} gpg2 --batch --passphrase '' --quick-add-key "${KEY_ID}" default sign 0
+        KEY_ID=$(HOME=${TMP_KEY_DIR} gpg2 --list-keys --with-colons "${KEY_NAME}"  | grep '^fpr:' | head -n 1 | cut -d : -f 10)
+        HOME=${TMP_KEY_DIR} gpg2 --batch --passphrase '' --quick-add-key "${KEY_ID}" default sign 0
     fi
 
     # export public and private key
-    HOME=${KEY_DIR} gpg2 --export -a "${KEY_NAME}" > "${KEY_DIR}/${KEY_NAME}-public"
-    HOME=${KEY_DIR} gpg2 --export-secret-keys -a "${KEY_NAME}" > "${KEY_DIR}/${KEY_NAME}-private"
+    HOME=${TMP_KEY_DIR} gpg2 --export -a "${KEY_NAME}" > "${KEY_DIR}/${KEY_NAME}-public"
+    HOME=${TMP_KEY_DIR} gpg2 --export-secret-keys -a "${KEY_NAME}" > "${KEY_DIR}/${KEY_NAME}-private"
```

Comment:
Can you please use ``TMP_KEY_DIR`` in all the cases in the above commands, so that it is consistent and one doesn't have to wander what is the difference between these two variables?